### ISSUE TITLE
Annotation or Notes Tracks

### ISSUE THREAD

**Issue by Obsyden** [LMMS/lmms#6188]
### Enhancement Summary

An annotation/notes track that can be added to the song-editor just like an instrument track, only holding text information and not contributing any audio or sound data. Hovering over 'bars' of this note track would open a mini-text editor similar to the project notes where the full note could be viewed and edited. Essentially, this is like being able to write notes in the margins of sheet music. Ideally, this would replace or complement the functionality of the Project Notes window.

#### Justification

This would be incredibly helpful for holding lyrics at a certain set of bars in a song, or indicating a key or chord change without having to look at the piano roll for instrument tracks.
I understand that the project notes exist, but I feel that it's a slow and confusing workflow (especially with lyrics) to write down "key change at bar 53", especially if the project is still being written and bar 53 may become bar 41 or something. 
Having notes act similarly to instrument tracks with their own bars in the song-editor will provide the writer with a better feel of where the text is in relation to the actual song; as opposed to writing all annotations in a separate window.

#### Mockup

Notes (or Annotation?) tracks can be added to the song-editor and look similar to any other type of track (only with no mute button obviously.) The bars present a small snippet of the text they contain (similar to the sample tracks showing the file-name.)
![Screenshot_2021-10-06_10-18-17](https://user-images.githubusercontent.com/41983520/136117250-0d42498e-8b09-40e9-9572-57e935e579a3.png)
Hovering over (or double-clicking?) a notes bar will open a mini text-editor interface to see and edit the full text.
![preview 2](https://user-images.githubusercontent.com/41983520/136117244-dc0b1450-4e83-4835-915e-cd9e7fbf470c.png)

**Comment by Obsyden** [comment]
Yes, the truncation is a good point; it's a waste to have so much space being taken by an ellipsis. Comment tracks is sort of similar to what I'm proposing here, it's a shame that there's no issue for it. #5929 I feel still suffers from the same problem as the current Project Notes - comments can't be tied to a specific point in time. As someone who writes songs with lyrics in them in LMMS, this would be a total game-changer for myself.

**Comment by Veratil** [comment]
Small related aside: I'd like to point out that part of Standard MIDI spec also allows text events so depending on how implemented we tie into event notes on the piano roll.

```
FF 01 len text Text Event
Any amount of text describing anything. It is a good idea to put a text event right at the beginning of a track,
 with the name of the track, a description of its intended orchestration, and any other information which the
 user wants to put there. Text events may also occur at other times in a track, to be used as lyrics, or
 descriptions of cue points. The text in this event should be printable ASCII characters for maximum interchange.
 However, other character codes using the high-order bit may be used for interchange of files between different
 programs on the same computer which supports an extended character set. Programs on a computer which
 does not support non-ASCII characters should ignore those characters. 
```

**Comment by anytizer** [comment]
Let us call it Annotation track.

The first change would be to able to add automation-track like instrument rack but without waiting for the automation controls to be attached. Limited features can be still achieved with the current automation track but without automation role assigned to it. ie. repurposing automation track as annotation track.

It works, because:

* it can be named, which can appear as lyric; assuming it works as short note.
* it can be colored. But probably no need to support italic text.
* Truncated text will try to appear in details when zoomed in.
* Multiple tracks can be added, without affecting the sound quality or change in .mmpz xml format.
* Probably supports Unicode too.
* Does not need new feature in LMMS. ;-)

But why new **~~automation~~ annotation track** is required?

* Because the automation track's name is susceptible to reset, due to which comments will be lost.
* It is always waiting for some control to be attached with it.
* The nature is not native to note-like but it is an automation track controller envelope.
* Long notes cannot be attached.
